implementations/rust/ockam/ockam_command/src/node/create.rs

```
@@ -239,27 +244,102 @@ async fn run_foreground_node(
         .await?;
     }
 
-    if let Some(authority_identities) = &cmd.authority_identities {
-        for auth in authority_identities.iter() {
-            let vault = Vault::create();
-            let i = PublicIdentity::import(auth.identity(), vault).await?;
-            cfg.authorities(&node_name)?
-                .add_authority(i.identifier().clone(), auth.clone())?;
-        }
-    }
-
     let project_id = match &cmd.project {
         Some(path) => {
             let s = tokio::fs::read_to_string(path).await?;
             let p: ProjectInfo = serde_json::from_str(&s)?;
             let project_id = p.id.to_string();
             project::config::set_project(cfg, &(&p).into()).await?;
-            add_project_authority_from_project_info(p, &node_name, cfg).await?;
             Some(project_id)
         }
         None => None,
     };
 
+    let mut trust_context_config = if let Some(authority_identity) = &cmd.authority_identity {
+        let vault = Vault::create();
+        let authority_public_identity =
+            PublicIdentity::import(&hex::decode(authority_identity)?, vault).await?;
+
+        let own_cred = match &cmd.credential {
+            Some(cred_name) => {
+                let state = opts.state.credentials.get(cred_name)?;
+                Some(CredentialRetrieverType::FromPath(state))
+            }
+            None => None,
```

This seems to be the only case where we can end-up with a `TrustAuthorityConfig` which would have no `own_credential` setup. Is it a legitimate case? Is that going to fail anyway later in the process? If that is the case, maybe we should throw an error right away to let the user fix the configuration (and then we can remove the `Option` in `TrustAuthorityConfig`)

---

No, it's an expected use case.

There's the mutual and one way credential exchange.

I know currently the orchestrator implementations do not have a credential to return.

---

Ok, thanks for the confirmation